src/init.cpp

```diff
@@ -1666,8 +1666,12 @@ bool AppInit2()
         }
 
         if (fFirstRun) {
-            // Create new keyUser and set as default key
-            CPubKey newDefaultKey;
+            // Create new HD Wallet
+            LogPrintf("Creating HD Wallet\n");
+            // Ensure this wallet.dat can only be opened by clients supporting HD.
+            pwalletMain->SetMinVersion(FEATURE_LATEST);
+            pwalletMain->SetupSPKM();
```

Comment:
This (or a more manual process) needs to also be done when using the `-upgradewallet` runtime option to actually upgrade a non-HD wallet to a HD wallet (as noted in the `sethdseed` RPC help texts).

the `if` block above `if (fFirstRun)` here only updates the wallet's versions, but doesn't setup any HD seed/chain

Reply:
Done. Just updated the PR description too.

The `-upgradewallet` flag will update previous wallet to HD and can be performed adding it at startup time or going to settings and pressing the Upgrade Wallet button.